### PR TITLE
Remove several unused packages

### PR DESCRIPTION
**Pull request by SyntaxColoring**
# Overview

These commits deselect a bunch of unused packages in an attempt to free up some filesystem space. (See https://github.com/Opentrons/opentrons/issues/8184.)

Unfortunately, I still don't think this is enough for a top-level `make push` to succeed, so https://github.com/Opentrons/opentrons/issues/8184 will stay open.

# Changelog

* Removed Python packages for JSON-RPC.
    * Per @sfoster:
      > it was a first attempt at making a nice network-transparent interface to the hardware controller for test engineering and others who want to directly interact with it
      >
      > i think we're moving away from the concept, you can remove it
    * Also removed `chardet`, a dependency of this.
* Removed [`pyserial-asyncio`](https://pyserial-asyncio.readthedocs.io/en/latest/shortintro.html). I couldn't find any instance of `import serial_asyncio` in Opentrons/opentrons.
* Removed `json-c`, a library that isn't mandated by anything else that we have installed.
* Removed a bunch of nginx modules that appeared unused to me, judging by our `nginx.conf`.
* Removed `sl`. 😢 

# Review requests

Do any of the removed packages strike you as something that we should definitely not remove?

### CODE REVIEW

**Comment by SyntaxColoring** [comment]
Ideas for further work:

* [This PR](https://github.com/Opentrons/buildroot/pull/86) added `git`, `curl`, and `wget`.

  Do we really need `git`? Do we really need both `curl` and `wget`, or can we just pick one?
* Busybox has its own implementations of `curl` and/or `wget` (can't remember which, exactly). So we have two implementations installed. Can we get rid of one? Are we accidentally doing this with any other commands?
* Should we strip test files from Python packages, sort of automating the workaround in Opentrons/opentrons#8184?

File: configs/ot2_defconfig [hunk 1]
```diff
@@ -91,7 +85,29 @@ BR2_PACKAGE_DROPBEAR=y
 BR2_PACKAGE_IW=y
 BR2_PACKAGE_NETWORK_MANAGER=y
 BR2_PACKAGE_NGINX=y
-BR2_PACKAGE_NGINX_HTTP_SSL_MODULE=y
+# BR2_PACKAGE_NGINX_HTTP_CHARSET_MODULE is not set
```
Comment (sfoster1) [inline configs/ot2_defconfig:88]: huh, stuff like this usually doesn't make it into a defconfig, wonder if nginx is doing something weird to parse these. would expect them to be `BR2_PACKAGE_NGINX_HTTP_GZIP_MODULE=n` and so on

**Review by sfoster1 — APPROVED** [review]
This looks good but I think there's more we can do like, 
- Strip out all the bluetooth stuff. this was a temporary experiment that at this point I think is not going to happen
- if busybox has a curl let's use that; wget can still be useful, but i don't really care whether it happens or not
- i do not remember at all why we added git, let's not
- i doubt we need screen or tmux but they're also tiny so whatever

Stripping as much non-critical content from the python stuff is absolutely a good idea. We may also find it useful to install them as gzipped wheels to save disk space.

File: configs/ot2_defconfig (hunk 1 above)
Comment (SyntaxColoring) [reply configs/ot2_defconfig:88]: Yeah, it is weird and mysterious.

**Comment by SyntaxColoring** [comment]
Aggregating those further ideas in #144.

**Pull request merged**